src/invariant/Invariant.h

```diff
@@ -25,6 +25,8 @@ class Invariant
 
     virtual std::string getName() const = 0;
 
+    virtual bool isStrict() const = 0;
```

Comment:
I think making this one virtual is a bit of overkill. What about mStrict field that is set in constructor and returned here?